### PR TITLE
Convert 'unsafeHint' to ghc-lib-parser

### PR DESCRIPTION
**Pull request by shayne-fletcher-da**
This PR converts `Unsafe.hs`.

### CODE REVIEW

**Comment by shayne-fletcher-da** [comment]
This is one is incontrovertible evidence that number of lines is a poor measure of code complexity 🤣

File: src/GHC/Util.hs [hunk 1]
```diff
@@ -197,6 +198,13 @@ isDot x
     , ident == mkVarUnqual (fsLit ".") = True
   | otherwise                          = False
 
+-- | 'isDol e'  if 'e' is the unqualifed variable '$'.
+isDol :: HsExpr GhcPs -> Bool
+isDol x
+  | HsVar _ (L _ ident) <- x
+    , ident == mkVarUnqual (fsLit "$") = True
+  | otherwise                          = False
```
Comment (ndmitchell) [inline src/GHC/Util.hs]: I'd have written this as:

```
isDol (HsVar _ (L _ ident)) = ident == mkVarUnqual (fsList "$")
isDol _ = False
```

**Comment by ndmitchell** [comment]
Thanks! You're welcome to PR a change to `isDol` or not bother - entirely your call.

**Comment by shayne-fletcher-da** [comment]
> Thanks! You're welcome to PR a change to `isDol` or not bother - entirely your call.

Right you are. Agree that's a little easier on the eye. See https://github.com/ndmitchell/hlint/pull/729

**Pull request merged**